src/js/components/Table/Paginated/PaginatedTable.tsx

```diff
@@ -76,11 +94,61 @@ const PaginatedTable: React.FunctionComponent<Props> = ({
     fetchData();
   }, [itemsPerPage, currentPage, sortName, sortAscending]);
 
-  const rows = dataSet.map(mapDataToRow);
+  let rows = dataSet.map(mapDataToRow);
+
+  let cols = columns;
+  if (rowsAreSelectable) {
+    cols = [{name: "selected", label: <div>
+      <Checkbox name="select-all" inputProps={{checked: selected.length != 0 && selected.length == rows.length, onClick: () => {
+        // if these aren't selected, we need to select them.
+        if (selected.length < rows.length) {
+          setSelected(rows);
+        } else {
+          // empty it out
+          setSelected([]);
+        }
+      }}}/>
+    </div>, mods: "u-size1of12"}, ...cols];
+
+      // use IDs as keys to determine uniqueness
+    const selectedids = selected.map(e => e.id);
+    rows = rows.map((ele) => Object.assign({}, ele, {selected: <div>
+      <Checkbox name={`select-${ele.id}`} inputProps={{checked: selectedids.includes(ele.id), onClick: () => {
+        if (selectedids.includes(ele.id)) {
+          setSelected(selected.filter(e => e.id != ele.id));
+        } else {
+          setSelected([...selected, ele]);
+        }
+      }}}/>
+    </div>}));
+  }
+
+  // collisions overwrite
+  const bulkActionFuncsByLabel = bulkActions?.reduce((acc, action: BulkAction) => {
+    acc[action.label] = action.onSelected;
```

Comment:
Are the bulk actions the only way to access the selected items?

Reply:
Yeah, thee selected items state is maintained by the table. If we want to expose other ways to manage that, we can.